irpc-derive/src/lib.rs

```diff
@@ -429,145 +386,136 @@ impl Parse for MacroArgs {
-    }
+}
 
-    /// Fail if there are any unknown arguments remaining
-    fn check_empty(&self, span: Span) -> syn::Result<()> {
-        if self.types.is_empty() {
-            Ok(())
-        } else {
-            Err(syn::Error::new(
-                span,
-                format!(
-                    "Unknown arguments provided: {:?}",
-                    self.types.keys().collect::<Vec<_>>()
-                ),
-            ))
-        }
-    }
+#[derive(Default)]
+struct RpcArgs {
+    rx: Option<Type>,
+    tx: Option<Type>,
 }
 
 /// Parse the rpc args as a comma separated list of name=type pairs
-impl Parse for NamedTypeArgs {
+impl Parse for RpcArgs {
     fn parse(input: ParseStream) -> syn::Result<Self> {
-        let mut types = BTreeMap::new();
-        let mut wrap = None;
-
-        loop {
-            if input.is_empty() {
-                break;
-            }
-
-            let key: Ident = input.parse()?;
-            if key == "wrap" {
-                let value = if input.peek(Token![=]) {
-                    let _: Token![=] = input.parse()?;
-                    let value: Ident = input.parse()?;
-                    Some(value)
-                } else {
-                    None
-                };
-                wrap = Some(value);
+        let mut this = Self::default();
+        while !input.is_empty() {
+            let arg: Ident = input.parse()?;
+            let _: Token![=] = input.parse()?;
+            let value: Type = input.parse()?;
+            if arg == RX_ATTR {
+                this.rx = Some(value);
+            } else if arg == TX_ATTR {
+                this.tx = Some(value);
             } else {
-                let _: Token![=] = input.parse()?;
-                let value: Type = input.parse()?;
-                types.insert(key.to_string(), value);
+                syn_err(arg.span(), "Unexpected argument in rpc attribute")?;
             }
-
             if !input.peek(Token![,]) {
                 break;
+            } else {
+                let _: Token![,] = input.parse()?;
             }
-            let _: Token![,] = input.parse()?;
         }
 
-        Ok(NamedTypeArgs { types, wrap })
+        Ok(this)
     }
 }
 
-fn type_from_ident(ident: Ident) -> Type {
+struct WrapArgs {
+    vis: Option<Visibility>,
+    ident: Ident,
+    derive: Vec<Type>,
+}
+
+impl Parse for WrapArgs {
+    fn parse(input: ParseStream) -> syn::Result<Self> {
+        let vis = match input.parse::<Visibility>()? {
+            Visibility::Inherited => None,
+            vis => Some(vis),
+        };
+        let ident: Ident = input.parse()?;
+        let mut this = Self {
+            ident,
+            derive: Default::default(),
+            vis,
+        };
+        while input.peek(Token![,]) {
+            let _: Token![,] = input.parse()?;
+            let arg: Ident = input.parse()?;
+            match arg.to_string().as_str() {
+                "derive" => {
```

Comment:
Can we just pass through all the things instead of specifically looking for derive(foo)? Maybe somebody wants to pass an attribute other than derive?

Reply:
Let's do this once the need arises, ok? Not passing through everything reserves us the chance to add more of our own attributes.